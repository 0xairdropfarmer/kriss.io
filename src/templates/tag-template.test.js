import React from 'react';
import renderer from 'react-test-renderer';
import TagTemplate from './tag-template';

describe('TagTemplate', () => {
  const props = {
    data: {
      allMarkdownRemark: {
        group: [
          {
            fieldValue: 'test_0',
            totalCount: 1
          },
          {
            fieldValue: 'test_1',
            totalCount: 2
          }
        ],
        edges: [
          {
            node: {
              fields: {
                slug: '/test_0',
                categorySlug: '/test'
              },
              frontmatter: {
                date: '2016-09-01T23:46:37.121Z',
                description: 'test_0',
                category: 'test',
                title: 'test_0'
              }
            }
          },
          {
            node: {
              fields: {
                slug: '/test_1',
                categorySlug: '/test'
              },
              frontmatter: {
                date: '2016-09-01T23:46:37.121Z',
                description: 'test_1',
                category: 'test',
                title: 'test_1'
              }
            }
          }
        ]
      },
      site: {
        siteMetadata: {
          title: 'test',
          subtitle: 'test'
        }
      }
    },
    pageContext: {
      tag: 'test',
      page: 1
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<TagTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
