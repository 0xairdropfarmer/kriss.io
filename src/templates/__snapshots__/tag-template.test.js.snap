// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <StaticQuery
    render={[Function]}
  />
  <div
    className="page"
  >
    <div
      className="page__inner"
    >
      <h1
        className="page__title"
      >
        test
      </h1>
      <div
        className="page__body"
      >
        <div
          className="feed"
        >
          <div
            className="feed__item"
          >
            <div
              className="feed__item-meta"
            >
              <time
                className="feed__item-meta-time"
                dateTime="September 2, 2016"
              >
                September 2016
              </time>
              <span
                className="feed__item-meta-divider"
              />
              <span
                className="feed__item-meta-category"
              >
                <Link
                  className="feed__item-meta-category-link"
                  to="/test"
                >
                  test
                </Link>
              </span>
            </div>
            <h2
              className="feed__item-title"
            >
              <Link
                className="feed__item-title-link"
                to="/test_0"
              >
                test_0
              </Link>
            </h2>
            <p
              className="feed__item-description"
            >
              test_0
            </p>
            <Link
              className="feed__item-readmore"
              to="/test_0"
            >
              Read
            </Link>
          </div>
          <div
            className="feed__item"
          >
            <div
              className="feed__item-meta"
            >
              <time
                className="feed__item-meta-time"
                dateTime="September 2, 2016"
              >
                September 2016
              </time>
              <span
                className="feed__item-meta-divider"
              />
              <span
                className="feed__item-meta-category"
              >
                <Link
                  className="feed__item-meta-category-link"
                  to="/test"
                >
                  test
                </Link>
              </span>
            </div>
            <h2
              className="feed__item-title"
            >
              <Link
                className="feed__item-title-link"
                to="/test_1"
              >
                test_1
              </Link>
            </h2>
            <p
              className="feed__item-description"
            >
              test_1
            </p>
            <Link
              className="feed__item-readmore"
              to="/test_1"
            >
              Read
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
